src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1195,17 +1197,56 @@ private ImplicitNamedTypeSymbol BindScriptClass()
             return namespaceOrType as ImplicitNamedTypeSymbol;
         }
 
-        internal Imports GlobalImports
+        /// <summary>
+        /// Global imports (including those from previous submissions, if there are any).
+        /// </summary>
+        internal Imports GlobalImports => _globalImports.Value;
+
+        private Imports BindGlobalImports() => Imports.FromGlobalUsings(this);
+
+        /// <summary>
+        /// Imports declared by this submission (null if this isn't one).
+        /// </summary>
+        internal Imports SubmissionImports
         {
-            get { return _globalImports.Value; }
+            get
+            {
+                if (!this.IsSubmission)
+                {
+                    return null;
+                }
```

Comment:
Consider changing SubmissionImports to a method and asserting this.IsSubmission.